### PR TITLE
Add xz, update MySQL and fix docs

### LINKED ISSUE #78: MySQL 8.0 support
Installation using fullInstall in [DIRAC/tests/Jenkins/dirac_ci.sh](https://github.com/DIRACGrid/DIRAC/blob/integration/tests/Jenkins/dirac_ci.sh) fails when the script wants to create a new DB in MySQL. It seems like MySQL 8 changes default authentication plugin from `mysql_native_password` to `caching_sha2_password`, as mentioned [here](https://stackoverflow.com/questions/49963383/authentication-plugin-caching-sha2-password).
Install log is here: [log.txt](https://github.com/DIRACGrid/DIRAC/files/3300247/log.txt).

The following environment variables were used:
```
DEBUG=True
DIRAC_RELEASE=v6r22-pre3
DB_USER=Dirac
DB_PASSWORD=Dirac
DB_ROOTUSER=admin
DB_ROOTPWD=password
DB_HOST=172.18.0.4
DB_PORT=3306
DIRACOSVER=v1r1

NoSQLDB_HOST=172.18.0.5
NoSQLDB_PORT=9200

repository=DIRACGrid
branch=rel-v6r22
DIRACBRANCH=rel-v6r22
```
as well as having MySQL 8.0 and ElasticSearch 6.6.0 in separate Docker containers. The server was also installed in a Docker container. The admin user in docker was created using

```
CREATE USER 'admin'@'%' IDENTIFIED BY 'password';
GRANT ALL PRIVILEGES ON *.* TO 'admin'@'%' WITH GRANT OPTION;
FLUSH PRIVILEGES;
quit
```

Solutions to this include:
1. Upgrading MySQL client in DIRACOS to 8.0 or higher and implement backwards compatibility for 5.X. (DIRACOS bundling + DIRAC code)
2. Changing authentication method for both Dirac and admin users using `ALTER USER admin
IDENTIFIED WITH mysql_native_password
BY 'password';` (DIRAC code)

### PR DESCRIPTION
**Pull request by chrisburr**
BEGINRELEASENOTES

NEW: Add xz for use with Python's backport.lzma
NEW: Add uproot and related compression libraries for reading ROOT files
CHANGE: Update MySQL from 5.7.21 to 5.7.28 for support with servers running MySQL 8

ENDRELEASENOTES

Fixes #78

### CODE REVIEW

**Comment by chaen** [comment]
I created a branch `xz` which will trigger a test build

**Comment by petricm** [comment]
Why is this needed in vanilla DIRACOS, and I don't mean the MySQL change?

**Comment by chrisburr** [comment]
The issue is that extensions currently don't support compiled code as far as I know. For the pip packages its easy to hack around, but `xz` is tricker.

**Comment by petricm** [comment]
The tests are failing for uproot

https://gitlab.cern.ch/CLICdp/iLCDirac/DIRACOS/pipelines/1274758

**Comment by chaen** [comment]
yes because it was not in the requirements.txt when the build started. I will rebuild now.

I will also open a task to allow for extensions with compilation

File: config/diracos.json [hunk 1]
```diff
@@ -639,6 +639,17 @@
                   "src": "https://diracos.web.cern.ch/diracos/SRPM/zlib-1.2.3-29.el6.src.rpm",
                   "name": "zlib"
                },
+               {
+                  "comment": "For backports.lzma",
```
Comment (chaen) [inline config/diracos.json]: Please just say that this is needed for uproot

**Comment by chaen** [comment]
The tests are working, and the bundle is even 7MB smaller.
If @petricm is ok as is, I will download the SRPM on EOS, you can replace the link and then we merge

**Comment by chaen** [comment]
One thing though, since we change the mysql version: please try running the DIRAC integration tests with it. You can install this specific version of DIRACOS in `dirac-install` with `--diracos-version` or something like that.  I think the jenkins integration jobs offer you that as a parameter already

**Comment by petricm** [comment]
Fine with me, add the SRPMs to the website

**Comment by chrisburr** [comment]
Yeah, the new mysql update is notably smaller.
(I should have included singularity in this PR so it only makes the bundle 5MB bigger 😆)

I'll run the integration tests shortly.

File: config/diracos.json [hunk 2]
```diff
@@ -656,7 +667,7 @@
                   "buildOnly": true
                },
                {
-                  "src": "https://diracos.web.cern.ch/diracos/SRPM/mysql-community-5.7.21-1.el6.src.rpm",
+                  "src": "http://repo.mysql.com/yum/mysql-5.7-community/el/6/SRPMS/mysql-community-5.7.28-1.el6.src.rpm",
```
Comment (petricm) [inline config/diracos.json]: I have downloaded this SRPM to the diracos webpage please use that location

File: config/diracos.json [hunk 3]
```diff
@@ -639,6 +639,17 @@
                   "src": "https://diracos.web.cern.ch/diracos/SRPM/zlib-1.2.3-29.el6.src.rpm",
                   "name": "zlib"
                },
+               {
+                  "comment": "For backports.lzma, required to read compressed files with uproot",
+                  "src": "http://vault.centos.org/6.10/os/Source/SPackages/xz-4.999.9-0.5.beta.20091007git.el6.src.rpm",
```
Comment (petricm) [inline config/diracos.json]: I have downloaded this SRPM to the diracos webpage please use that location

**Comment by chrisburr** [comment]
The tests have passed in for a MySQL 5.7 server:
* For `rel-v7r0`: [here](https://github.com/chrisburr/DIRAC/runs/339714957)
* For `rel-v7r1`:  [here](https://github.com/chrisburr/DIRAC/runs/339729241)

(MySQL 8.0 fails for `rel-v7r0` but I presume that's a known DIRAC issue)

**Comment by fstagni** [comment]
MySQL 8.0 is supported only from `rel-v7r1` indeed.

**Comment by chrisburr** [comment]
Perfect, once a new DIRACOS release is made it can be added to the test matrix by uncommenting [this line](https://github.com/DIRACGrid/DIRAC/blob/integration/.github/workflows/integration.yml#L16).

**Pull request merged**